openedx/core/djangoapps/content_libraries/management/commands/content_libraries_import.py

```diff
@@ -0,0 +1,376 @@
+                oauth_key, oauth_secret = options['oauth_creds']
+            else:
+                raise CommandError("Method 'remote' requires one of the "
+                                   "--oauth-* options, and none was specified.")
+            edx_client = EdxApiClient(
+                options['lms_url'],
+                options['studio_url'],
+                oauth_key,
+                oauth_secret)
+        elif options['method'] == 'modulestore':
+            edx_client = EdxModulestoreClient()
+        else:
+            assert False, f"Method not supported: {options['method']}"
+
+        # Query the course and rerieve all course blocks.
+
+        export_keys = edx_client.get_export_keys(options['course-key'])
+        if not export_keys:
+            raise CommandError("The courseware course specified does not have "
+                               "any exportable content.  No action take.")
+
+        # Import each block, skipping the ones that fail.
+
+        failed_blocks = []
+        for index, block_key in enumerate(export_keys):
+            try:
+                self.stdout.write(f"{index + 1}/{len(export_keys)}: {block_key}: ", ending='')
+                self.import_block(edx_client, library, block_key)
+            except Exception as exc:  # pylint: disable=broad-except
+                self.write_error('❌')
+                self.stderr.write(f"Failed to import modulestore block: {exc}")
+                log.exception("Error importing modulestore block: %s", block_key)
+                failed_blocks.append(block_key)
+                continue
+            else:
+                self.write_success('✓')
+        if failed_blocks:
+            self.write_error(f"❌ {len(failed_blocks)} out of {len(export_keys)} failed:")
+            for key in failed_blocks:
+                self.write_error(str(key))
+
+    def import_block(self, edx_client, library, modulestore_key):
+        """
+        Import a single modulestore block.
+        """
+
+        # Get or create the block in the library.
+
+        block_data = edx_client.get_block_data(modulestore_key)
+
+        try:
+            library_block = contentlib_api.create_library_block(
+                library.key,
+                modulestore_key.block_type,
+                modulestore_key.block_id)
+            blockstore_key = library_block.usage_key
+        except contentlib_api.LibraryBlockAlreadyExists:
+            blockstore_key = LibraryUsageLocatorV2(
+                lib_key=library.key,
+                block_type=modulestore_key.block_type,
+                usage_id=modulestore_key.block_id,
+            )
+            contentlib_api.get_library_block(blockstore_key)
+
+        # Handle static files.
+
+        for filename, static_file in block_data.get('static_files', {}).items():
+            files = [
+                f.path for f in
+                contentlib_api.get_library_block_static_asset_files(blockstore_key)
+            ]
```

Comment:
It seems inefficient to get all static asset files each time through this loop - but I understand it makes for simpler code so 🤷🏼‍♀️ .

Reply:
I am not aware of a bulk API to retrieve static files metadata from blockstore bundles. Indeed, that would be a great follow up for this for massive course imports (e.g. > 10K blocks). I am glad you seem inclined to agree that this can be done as a follow-up while keeping this first version simpler.